tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -582,7 +583,231 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
+                        "because it both completes the original stage and requests the result. The withContextCapture " +
+                        "method does not allow new async threads to be allocated even when the *Async methods are used.");
+
+                Buffer.get().append("-stage4");
+                Buffer.set(new StringBuffer("withContextCapture-CompletionStage-test-buffer-D"));
+
+                return i - 2345;
+            });
+
+            Buffer.set(new StringBuffer("withContextCapture-CompletionStage-test-buffer-E"));
+
+            unmanagedStage1.complete(1234);
+
+            CompletableFuture<Integer> cf4 = stage4.toCompletableFuture();
+
+            Assert.assertEquals(cf4.getNow(987), Integer.valueOf(123),
+                    "Completion stage created by withContextCapture did not complete with same value as original stage.");
+
+            Assert.assertEquals(buffer.toString(), "withContextCapture-CompletionStage-test-buffer-A-stage3-stage4");
+
+            // Has context been properly restored after the contextual operation(s)?
+            Assert.assertEquals(Buffer.get().toString(), "withContextCapture-CompletionStage-test-buffer-E",
+                    "Previous context was not restored after context was propagated for contextual action.");
+            Assert.assertEquals(Label.get(), "withContextCapture-CompletableFuture-test-label",
+                    "Previous context was not restored after context was cleared for contextual action.");
+        }
+        finally {
+            // Restore original values
+            Buffer.set(null);
+            Label.set(null);
+        }
+    }
+
+    /**
+     * Verify that dependent stages created via withContextCapture can be completed independently
+     * of the original stage.
+     */
+    public void withContextCaptureDependentStageForcedCompletion() throws ExecutionException, InterruptedException {
+        ThreadContext contextPropagator = ThreadContext.builder().build();
+
+        CompletableFuture<String> stage1 = new CompletableFuture<String>();
+        CompletableFuture<String> stage2 = contextPropagator.withContextCapture(stage1);
+
+        Assert.assertTrue(stage2.complete("stage_2_done"),
+                "It should be possible to complete a CompletableFuture created via withContextCapture without completing the original stage.");
+
+        Assert.assertFalse(stage1.isDone(),
+                "Completion of the dependent stage must not imply completion of the original stage.");
+
+        Assert.assertTrue(stage1.complete("stage_1_done"),
+                "It should be possible to complete the original stage with a different result after dependent stage was forcibly completed.");
+
+        Assert.assertEquals(stage1.get(), Integer.valueOf("stage_1_done"),
+                "Completion stage result does not match the result with which it was forcibly completed.");
+
+        Assert.assertEquals(stage2.get(), Integer.valueOf("stage_2_done"),
```

Comment:
Did you intend to use Integer.valueOf() here?

Reply:
Fixing, the same as above